src/directLineStreaming.ts

```diff
@@ -0,0 +1,345 @@
+// In order to keep file size down, only import the parts of rxjs that we use
+
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Observable } from 'rxjs/Observable';
+import { Subscriber } from 'rxjs/Subscriber';
+import * as BFSE from 'botframework-streaming';
+
+import {
+  Activity,
+  ConnectionStatus,
+  Conversation,
+  DirectLine,
+  DirectLineOptions,
+  IBotConnection,
+  Media,
+  Message
+  } from './directLine';
+
+const DIRECT_LINE_VERSION = 'DirectLine/3.0';
+const MAX_RETRY_COUNT = 3;
+const refreshTokenLifetime = 30 * 60 * 1000;
+const timeout = 20 * 1000;
+const refreshTokenInterval = refreshTokenLifetime / 2;
+
+class StreamHandler implements BFSE.RequestHandler {
+  private connectionStatus$;
+  private subscriber: Subscriber<Activity>;
+  private activityQueue: Array<Activity> = [];
+
+  constructor(s: Subscriber<Activity>, c$: Observable<ConnectionStatus>) {
+    this.subscriber = s;
+    this.connectionStatus$ = c$;
+  }
+
+  public setSubscriber(s: Subscriber<Activity>) {
+    this.subscriber = s;
+  }
+
+  async processRequest(request: BFSE.IReceiveRequest, logger?: any): Promise<BFSE.StreamingResponse> {
+    let stream0 = request.streams.shift();
+    let activitySetJson = await stream0.readAsString();
+    let activitySet = JSON.parse(activitySetJson);
+
+    if (activitySet.activities.length !== 1) {
+      // Only one activity is expected in a set in streaming
+      this.subscriber.error(activitySet)
+      let r = new BFSE.StreamingResponse();
+      r.statusCode = 500;
+      return r;
+    }
+
+    var attachments = activitySet.activities[0].attachments;
+    let stream: BFSE.ContentStream;
+    while (stream = request.streams.shift()) {
+      let atch = await stream.readAsString();
+      var dataUri = "data:text/plain;base64," + atch;
+      attachments.push({ contentType: stream.contentType, contentUrl: dataUri });
+    }
+
+    let activity = activitySet.activities[0];
```

Comment:
We should put this line above. There are already some code looking for the `firstActivity`.